enhancements/storage/storage-security-policy.md

```diff
@@ -0,0 +1,198 @@
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+We want to allow Openshift admins to change `fsGroupChangePolicy` defaulting policies for certain namespaces so as pods in the given namespace can start faster.
+
+We also want to allow Openshift admins to set `selinuxChangePolicy` for certain namespaces, so as they can opt-out of potentially breaking changes being introduced via `SELinuxMount` feature upcoming in future k8s releases.
+
+## Motivation
+
+### Motivation for fsgroupchangepolicy defaulting
+
+Currently it can take a long time for volume's permission to be changed to match pod's `fsGroup`. This is a well known problem
+in Kubernetes and Openshift. See - https://access.redhat.com/solutions/6221251 for more details.
+
+To solve this problem, we introduced `fsGroupChangePolicy` - https://github.com/kubernetes/kubernetes/blob/b5608aea94cfb54fea3a63e1d74235759d036c51/pkg/apis/core/types.go#L3977 , which
+can be configured in a `PodSpec` and be set to `OnRootMismatch`, which instructs kubelet to stop recursively changing permissions of a volume, if top level directory permissions match as expected by kubelet.
+
+This usually speeds up pod startup because kubelet does not have to recursively change permission of each file and directory in the volume. This mechanism relies on heuristic that gets applied when recursively changing permissions of a volume -  kubelet changes permissions of top level directory as a last step in the entire process, when it performs full recursive `chown` and `chmod` of the volume.
+
+Having said that, `fsGroupChangePolicy` still defaults to `Always`, which means kubelet will always recursively change permissions. The reason for defaulting to `Always` is - mostly backward compatibility and if users bring in volumes that were used outside kubernetes. In these cases `OnRootMismatch` may not be a safe default.
+
+But for most intent and purpose, it can be a reasonable default and hence this proposal allows admins to configure a namespace such that, all pods created in the namespace will default to `OnRootMismatch` policy if pod does not specify a policy of its own. If a pod specifies its own `fsGroupChangePolicy` that is still used as a default and will not be overridden.
+
+### Motivation for selinuxChangePolicy defaulting
+
+https://github.com/kubernetes/enhancements/issues/1710 is bringing changes which can be breaking for certain pods which use different selinux contexts for same volume between different pods or same pod.
+
+This will speed up time spent while recursively changing selinux policies of a volume.
+
+By allowing cluster-admins to set namespace default of `selinuxChangePolicy` we want to give control over how these changes are applied to Openshift clusters, without explicitly  changing pods themselves.
```

Comment:
Similar to above, if we are making a breaking change, do we want to recommend the new, more secure(?) default, for new installations?

Reply:
So future k8s will use selinux mount option as the new default, which is faster option but could break some users ( @jsafrane has detailed data and metrics about clusters that could be broken). From security point of view, it is all the same. These options are performance optimization. 

I think the k8s default is a sensible option and we will document it, when it goes GA.

Reply:
Upstream, [KEP 1710](https://github.com/kubernetes/enhancements/tree/master/keps/sig-storage/1710-selinux-relabeling#story-3-cluster-upgrade) has an explanation how to upgrade with the breaking change safely.

This enhancement makes it easier for OCP users.

(My number show that <1% of clusters will be affected and for vast majority of them it's just a handful of Pods)

Reply:
So to confirm, new openshift clusters will use selinux by default? In which version is that going to happen? Should we document that somewhere explicitly in here?